libphal/phal_dump.C

```diff
@@ -131,6 +136,38 @@ struct pdbg_target* getProcFromFailingId(const uint32_t failingUnit)
 	return proc;
 }
 
+/**
```

Comment:
do not move the method like this, if you want to use the method above where it is used then do forward declaration

like 
void writeDumpFile(char* data, size_t len, std::filesystem::path& dumpPath);

at line no 139, this way it will not complain

Reply:
changed the code. Not using writeDumpFile.